### PR TITLE
Julia examples

### PR DESCRIPTION
**Pull request by Enet4**
This PR adds Julia to the omnibus. All examples were implemented, even the hidden `vector_return`. Feel free to scrutinise the code and text.

### CODE REVIEW

File: examples/Makefile [hunk 1]
```diff
@@ -79,5 +80,6 @@ python:
 haskell:
 nodejs:
 csharp:
+julia:
 
 .PHONY: c ruby python haskell nodejs csharp
```
Comment (shepmaster) [inline examples/Makefile]: Here too, right?

File: examples/tests_all.mk [hunk 2]
```diff
@@ -4,3 +4,4 @@ include ${TEST_PYTHON}
 include ${TEST_HASKELL}
 include ${TEST_NODEJS}
 include ${TEST_CSHARP}
+include ${TEST_JULIA}
```
Comment (shepmaster) [inline examples/tests_all.mk]: Maybe a newline to be consistent

**Comment by Enet4** [comment]
Oops, the part where docker would install Julia was missing. It's in now.

File: Dockerfile [hunk 3]
```diff
@@ -30,6 +30,15 @@ RUN \
         \
         && rm -rf /var/lib/apt/lists/*
 
+# Install Julia from the official website
+
+RUN curl https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.1-linux-x86_64.tar.gz --output julia-1.0.1-linux-x86_64.tar.gz
+RUN echo '9ffbcf7f4a111e13415954caccdd1ce90b5c835cee9f62d6ac708f5b752c87dd  julia-1.0.1-linux-x86_64.tar.gz' > .sha256
+RUN sha256sum -c .sha256
+RUN rm .sha256
+RUN tar xvf julia-1.0.1-linux-x86_64.tar.gz -C /root
```
Comment (shepmaster) [inline Dockerfile]: For Docker space-saving reasons, downloading the tarball and deleting it need to be in a single `RUN` statement. Something like:

```
RUN \
        curl \
        https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.1-linux-x86_64.tar.gz \
        --output julia-1.0.1-linux-x86_64.tar.gz \
        \
        && echo '9ffbcf7f4a111e13415954caccdd1ce90b5c835cee9f62d6ac708f5b752c87dd  julia-1.0.1-linux-x86_64.tar.gz' > .sha256 \
        && sha256sum -c .sha256 \
        && rm .sha256 \
        && tar xvf julia-1.0.1-linux-x86_64.tar.gz -C /root \
        && rm julia-1.0.1-linux-x86_64.tar.gz \
```

I hate it, but it's easier than having separate build files or using build stages.

File: Dockerfile [hunk 4]
```diff
@@ -21,7 +21,7 @@ RUN \
         \
         && apt-get install -y \
         ghc-8.4.3 \
-        mono-devel='5.14.*' \
+        mono-devel='5.16.*' \
```
Comment (shepmaster) [inline Dockerfile]: This should probably be a separate commit in a separate PR.

Comment (Enet4) [reply Dockerfile]: You're right, sorry. This was the only way to have it work locally, but then I rushed the commit a bit. :s

File: Dockerfile (hunk 3 above)
Comment (Enet4) [reply Dockerfile]: That was exactly what I was about to try next! It'll be but a moment...

**Comment by shepmaster** [comment]
Yep, this is super weird. The file is empty:

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
julia-1.0.1-linux-x86_64.tar.gz: empty
-rw-r--r-- 1 root root 0 Nov  4 15:05 julia-1.0.1-linux-x86_64.tar.gz
julia-1.0.1-linux-x86_64.tar.gz: FAILED
sha256sum: WARNING: 1 computed checksum did NOT match
```

**Comment by shepmaster** [comment]
Ugh. Locally:

```
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 200
< x-amz-id-2: vbwY5xL825xJm/POV/rvTrQKl5RE5ttoeyWFZ5+OzKzXGTaa6AIenBhvPM6HOWDTZDYrpkVPPbw=
< x-amz-request-id: 6FC5DA9116FC0022
< last-modified: Sat, 06 Oct 2018 11:43:05 GMT
< etag: "061f7187bf257c259af8d44fe1c5e88a-6"
< cache-control: max-age=86400, stale-if-error=86400, stale-while-revalidate=86400
< content-type: application/octet-stream
< server: AmazonS3
```

In Travis:

```
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
} [5 bytes data]
< HTTP/2 302 
< server: Varnish
< retry-after: 0
< location: https://storage.googleapis.com/julialang2/bin/linux/x64/1.0/julia-1.0.1-linux-x86_64.tar.gz
< accept-ranges: bytes
< date: Sun, 04 Nov 2018 15:56:32 GMT
< via: 1.1 varnish
< x-served-by: cache-sea1047-SEA
< x-cache: HIT
< x-cache-hits: 0
< x-timer: S1541346993.702208,VS0,VE0
< content-length: 0
```

Now, where might that be happening.

**Comment by shepmaster** [comment]
TL;DR: add `-L` to your curl command to follow redirects. That should get over that hump.

**Comment by shepmaster** [comment]
Thanks!
